src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -608,7 +613,8 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
         }
 
         internal static async Task InvokeAsync(IFunctionInvoker invoker, object[] invokeParameters, CancellationTokenSource timeoutTokenSource,
-            CancellationTokenSource functionCancellationTokenSource, bool throwOnTimeout, TimeSpan timerInterval, IFunctionInstance instance)
+            CancellationTokenSource functionCancellationTokenSource, bool throwOnTimeout, TimeSpan timerInterval, IFunctionInstance instance, ILogger logger = null,
+            JobHost jobHost = null, JobHostConfiguration config = null, IReadOnlyList<string> parameterNames = null)
```

Comment:
One workaround is to have 2 signatures. One with a required parameter (what Mathew wanted), and then one without the parameter that just passes null. The benefit of that is then we know statically which is which. 

---
In reply to: [124422411](https://github.com/Azure/azure-webjobs-sdk/pull/1207#discussion_r124422411) [](ancestors = 124422411)